### PR TITLE
Consider arbitrary shed

### PR DESCRIPTION
**Pull request by pcm32**
Uses the defined shed in the tool file, if available.

### CODE REVIEW

File: scripts/update-tool.py [hunk 1]
```diff
@@ -32,9 +36,17 @@ def update_file(fn, owner=None, name=None, without=False):
             continue
 
         logging.info("Fetching updates for {owner}/{name}".format(**tool))
+        
+        if 'tool_shed_url' in tool and tool['tool_shed_url'] in ts:
```
Comment (pinin4fjords) [inline scripts/update-tool.py]: So the idea here is to only create the test and main tool sheds in their standard locations once, but to allow custom tool sheds where required?

File: scripts/update-tool.py [hunk 2]
```diff
@@ -32,9 +36,17 @@ def update_file(fn, owner=None, name=None, without=False):
...
+            toolshed = ts[tool['tool_shed_url']]
+        elif 'tool_shed_url' in tool and not tool['tool_shed_url'] in ts:
+            ts[tool['tool_shed_url']] = toolshed.ToolShedInstance(url='https://'+tool['tool_shed_url'])
+            toolshed = ts[tool['tool_shed_url']]
+        else:
+            toolshed = ts[main]
```
Comment (pinin4fjords) [inline scripts/update-tool.py]:
```suggestion
        if 'tool_shed_url' in tool:
          if not tool['tool_shed_url'] in ts:
            ts[tool['tool_shed_url']] = toolshed.ToolShedInstance(url='https://'+tool['tool_shed_url'])
          toolshed = ts[tool['tool_shed_url']]  
        else:
          toolshed = ts[main]
```

**Review by pinin4fjords — APPROVED** [review]
Looks good- just a suggestion on logic simplification

File: scripts/update-tool.py (hunk 1 above)
Comment (pcm32) [reply scripts/update-tool.py]: Yes

**Comment by pcm32** [comment]
Thanks for the review and suggestions.

**Pull request merged**